src/components/CheckForm/AlertsPerCheck/FailedExecutionsAlert.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useCallback, useMemo } from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+import { durationToMilliseconds, parseDuration } from '@grafana/data';
+import { Checkbox, Icon, PopoverContent, Select, Stack, Tooltip } from '@grafana/ui';
+import { getTotalChecksPerPeriod } from 'checkUsageCalc';
+
+import { CheckAlertType, CheckFormValues } from 'types';
+
+import { useCheckFormContext } from '../CheckFormContext/CheckFormContext';
+import { ALERT_PERIODS, PredefinedAlertInterface } from './AlertsPerCheck.constants';
+import { ThresholdSelector } from './ThresholdSelector';
+
+export const FailedExecutionsAlert = ({
+  alert,
+  selected,
+  onSelectionChange,
+  tooltipContent,
+}: {
+  alert: PredefinedAlertInterface;
+  selected: boolean;
+  onSelectionChange: (type: CheckAlertType) => void;
+  tooltipContent: PopoverContent;
+}) => {
+  const { isFormDisabled } = useCheckFormContext();
+  const { getValues, control } = useFormContext<CheckFormValues>();
+
+  const handleToggleAlert = (type: CheckAlertType) => {
+    onSelectionChange(type);
+  };
+
+  const checkFrequency = getValues('frequency');
+  const probes = getValues('probes');
+
+  //min time range >= check frequency
+  const convertPeriodToSeconds = useCallback(
+    (period: string) => durationToMilliseconds(parseDuration(period)) / 1000,
+    []
+  );
+
+  const validPeriods = useMemo(
+    () => ALERT_PERIODS.filter((period) => convertPeriodToSeconds(period.value) >= checkFrequency),
+    [checkFrequency, convertPeriodToSeconds]
+  );
+
+  const defaultPeriod = validPeriods[0];
+  const period = getValues(`alerts.${alert.type}.period`) || defaultPeriod.value;
+
+  const testExecutionsPerPeriod = useMemo(() => {
+    if (!period) {
+      return '';
+    }
+    return getTotalChecksPerPeriod(probes.length, checkFrequency, convertPeriodToSeconds(period));
+  }, [checkFrequency, period, probes, convertPeriodToSeconds]);
+
+  const customTooltipContent: PopoverContent = (
+    <div>
+      The number of test executions is an approximation based on the check&apos;s frequency and the period chosen for
+      this alert.
+      {tooltipContent as React.ReactNode}
+    </div>
+  );
+
+  return (
+    <Stack alignItems="center">
+      <Stack alignItems="center">
+        <Checkbox
+          id={`alert-${alert.type}`}
+          data-testid={`checkbox-alert-${alert.type}`}
+          onClick={() => handleToggleAlert(alert.type)}
+          checked={selected}
+        />
+        <Stack alignItems="center">
+          Alert if <ThresholdSelector alert={alert} selected={selected} />
+          of {testExecutionsPerPeriod} tests fail in the last
```

Comment:
The PR description mentions that this alert is phrased as:

> Alert if at least x of n tests fail in the last T min

Which matches how the alert/threshold behaves, but this does not seem to comply with that.
Should we adopt that sentence?

Reply:
Fixed it 👍